### ISSUE TITLE
Controlling gutter position?

### ISSUE THREAD

**Issue by iamtommcc** [ezekg/flint#43]
So I've been playing around with Flint, and totally fallen in love with the syntax. Just. awesome.

One thing that irks me though is the fact that gutters are split between left and right margins -- meaning that the first and last elements in a row have a fairly redundant left and right margin respectively.

Is there an option or method to make gutters made only using right-margins (with the last-child having a margin of 0)? This is how quite a few other grid systems work by default and is often a bit more preferable/intuitive. 

For example: it would mean that a 12 column element in a 12 column grid would be able to take up the FULL max width, rather than the max width minus ~1% or whatever the gutter margins are.

See the first example in this link as to what I mean:
http://neat.bourbon.io/examples/.

**Comment by ezekg** [comment]
You can do that with [modifiers](http://flint.gs/docs/#modifiers). To make a row without gutters, you would do,

```scss
@include _(16 12 8 4, $gutter: none)
```

or to make a row of columns using `:nth-of-type` to modify the gutters,

```html
<section>
  <div>Item 1</div>
  <div>Item 2</div>
  <div>Item 3</div>
  <div>Item 4</div>
  <div>Item 5</div>
  <div>Item 6</div>
  <div>Item 7</div>
  <div>Item 8</div>
</section>
```
```scss
section {
  @include _(container, clear);
}

div {
  @include _(4);
  background: ghostwhite;
  padding: 2rem;
  margin-top: 0.75rem;
  margin-bottom: 0.75rem;
  
  @include _(desktop) {
    
    &:nth-of-type(4n+1) {
      @include _(4, $gutter: alpha);
      background: blue;
    }
    
    &:nth-of-type(4n+4) {
      @include _(4, $gutter: omega);
      background: red;
    }
  }
  
  @include _(laptop) {
    
    &:nth-of-type(3n+1) {
      @include _(4, $gutter: alpha);
      background: blue;
    }

    &:nth-of-type(3n+3) {
      @include _(4, $gutter: omega);
      background: red;
    }
  }
  
  @include _(tablet) {
    
    &:nth-of-type(2n+1) {
      @include _(4, $gutter: alpha);
      background: blue;
    }
    
    &:nth-of-type(2n+2) {
      @include _(4, $gutter: omega);
      background: red;
    }
  }
}
```

results in,
![image](https://cloud.githubusercontent.com/assets/6979737/12868246/329bb988-ccc8-11e5-8e71-1b1a98a610e3.png)

Here's a gist so that you can visually see what I mean and edit the code, http://www.sassmeister.com/gist/1e5e7025ebbdaa8e41d3. Let me know if that answers your question!

**Comment by ezekg** [comment]
Closing because of non-issue. Let me know if you need any further help or clarification.

**Comment by ezekg** [comment]
@iamtommcc, did you ever get a chance to look at this?

**Comment by iamtommcc** [comment]
Hey @ezekg, thanks a million for the super detailed response, really appreciate the effort :ok_hand: . That is exactly the functionality I'm talking about.

> The left/right gutters aren't really redundant, because they're designed to be placed inside of a container that doesn't have gutters. Every design (well, most) leaves margin on the left/right of page, so having gutters on both sides by default handles doing that for you

Totally true. Although the main thing I noticed is that, since the gutters are fluid, this edge-of-page margin gets too small to be of much benefit when you get down to small screen sizes (unless you set ungodly huge gutters).

I mean obviously, one can always manually add a small static px margin at small sizes to work around this... but considering I would end up doing this for most Flint projects, it kind of defeats the purpose of the left/right gutters existing in the first place.

Basically it would be awesome to have the left/right gutters as a global toggleable option. As this is a pretty small & insignificant gripe, it might not be worth the work, but still!

**Comment by ezekg** [comment]
The gutters are calculated separately for every breakpoint, so at `20em`, the gutters will be exactly `1em`; that's a perfectly normal gutter size. Screens don't get much smaller than that, and even if they did, you could bump down the lowest breakpoint or increase the gutter width a bit to account for that. I've never had an issue with gutters being too small.

**Comment by simonlayfield** [comment]
Hey @ezekg,

Just wanted to say I was tinkering with this kind of functionality in my own project and thought I'd see if there were any issues raised on here with a (more) intelligent solution (than my own).

...which it appears there is.

The level of detail in your responses to issues are such a massive help.

Keep up the good work man. Appreciate it.

**Comment by ezekg** [comment]
Thanks, @simonlayfield! Let me know if there's anything I can help you with.
